src/main/java/fact/io/hdureader/ZFITSHeapReader.java

```diff
@@ -244,57 +288,76 @@ private void unsmooth(short[] data){
      * @param buffer the coplet tile buffer (with blockheaders, but without the tile headers.)
      * @throws IOException in case the buffer ends before all bytes are read
      */
-    private short[] huffmanDecompression(ByteBuffer buffer) throws IOException {
+    private short[] huffmanDecompression(ByteBuffer buffer, int numRows, int sizeSingleRow) throws IOException {
```

Comment:
Not quite sure what this all does now. However I get the feeling this does the same for many rows instead of one. I do think its nicer when the loop happens outside of this function. Any reason why that won't work?

Reply:
It will work it is just that due to the way stuff was implemented before, this was the result, basically I would need to split the Huffman up a tiny little bit at the beginning. Will change this.

Reply:
fixed, now there is a new blockHuffmanDecompression that calls the huffmandecompression function for each row.